pymc_marketing/mmm/base.py

```diff
@@ -436,8 +437,13 @@ def plot_posterior_predictive(
         else:
             fig = ax.figure
 
-        if add_hdi:
-            for hdi_prob, alpha in zip((0.94, 0.50), (0.2, 0.4), strict=True):
+        if hdi_list is None:
+            hdi_list = [0.94]
+
+        if hdi_list:
+            # skipped if hdi_list is empty
+            alpha_list = np.linspace(0.2, 0.4, len(hdi_list))
+            for hdi_prob, alpha in zip(hdi_list, alpha_list, strict=True):
```

Comment:
Doesn't the `strict` cause an issue here? Should it be set to `False`?

Reply:
+1

Reply:
So the `strict=True` was there when I got here :)

It doesn't cause any issues because the output of the `np.linspace` for the alphas is equal to the number of hdi's.

![Screenshot 2024-10-24 at 11 35 37](https://github.com/user-attachments/assets/b5db30f9-406b-45bb-b339-4894d90d5e80)


There's no combination which causes it to fail.
![Screenshot 2024-10-24 at 11 36 16](https://github.com/user-attachments/assets/a52bef70-a5ae-4589-af53-09bef5c14129)

So probably makes sense to remove the `strict` kwarg as there's no way for the list of hdi's and alpha's to be of different lengths. Though you could argue that leaving it in there would be good for defensive programming when someone makes more changes to the function in the future. happy to do whatever -

Reply:
Ah didnt see using len of list